New landing page based on PyData-Sphinx-Theme (#2361)
helmholtz-analytics/heat: doc/source/index.rst

-----

.. Why Heat?
.. =========

.. Heat is a distributed tensor framework built on **PyTorch** and **mpi4py**. It provides highly optimized algorithms and data structures for tensor computations using CPUs, GPUs (CUDA/ROCm), and distributed cluster systems. It is designed to handle **massive arrays** that exceed the memory and computational limits of a single machine.

.. * **Seamless integration:** Port existing NumPy/SciPy code to multi-node clusters with minimal effort.
.. * **Hardware-agnostic:** Supports CPUs and GPUs (CUDA, ROCm, Apple MPS).
.. * **Efficient scaling:** Exploit the entire, cumulative RAM of your cluster for memory-intensive operations.

In a nutshell
===============

.. grid:: 1 1 2 2
   :gutter: 4
   :class-container: mt-4 mb-4

   .. grid-item::
      :columns: 12 12 5 5

      .. **Prototype on your laptop, execute on any cluster**

      Heat builds on **PyTorch** and **mpi4py** to process **massive arrays** - huge collections of images, high-dimensional climate simulation grids, or massive machine learning feature matrices - that exceed the memory and computational limits of a single machine.

>       Define your data distribution axis via the ``split`` parameter, assign hardware using the ``device`` attribute, and let Heat orchestrate data movement and cross-node communication.

Reviewer: ```suggestion
      Define your data distribution axis via the ``split`` parameter, assign hardware using the ``device`` attribute, and let Heat orchestrate the parallel computation.
```

      **Prototype locally, execute on any cluster.**

      You got the expensive compute, but multi-GPU runs seem out of reach?
      ``pip install heat``

> 

Reviewer: ```suggestion
```
I think this is a bit too much :D

Author: Too much reality!

   .. grid-item::
      :columns: 12 12 7 7

      .. code-block:: python
         :caption: my_script.py

         import heat as ht

         # Distributed random matrix generation
         A = ht.random.randn(40000, 10000, split=0, device="gpu")
         B = ht.random.randn(10000, 40000, split=1, device="gpu")

>          # Hardware-accelerated matrix multiplication

Reviewer: ```suggestion
         # Multi-GPU-accelerated matrix multiplication
```

         C = ht.matmul(A, B)

      .. code-block:: bash
         :caption: Run locally or scale across cluster nodes via MPI

         mpirun -np 4 python my_script.py

-----

Getting started
================

.. grid:: 1 2 3 3
   :gutter: 3
   :class-container: text-center mt-4 mb-4

   .. grid-item-card::
      :class-card: sd-card
      :link: /quick_start
      :link-type: doc

      .. image:: _static/images/install_graphics.png
         :alt: Installation
         :align: center
         :height: 140px

      .. div:: mt-3 **Installation**

      .. div:: text-muted mt-1

         **Install guide** from your laptop to HPC systems

   .. grid-item-card::
      :class-card: sd-card
      :link: /coverage_tables
      :link-type: doc

      .. image:: _static/images/tutorial_split_dndarray.svg
         :alt: Installation
         :align: center
         :height: 140px

      .. div:: mt-3 **NumPy API**

      .. div:: text-muted mt-1

>          **Numerical data processing**: NumPy/SciPy API compatibility tracking

Reviewer: It's a bit confusing that this is not a link, but seems clickable.

Author: good catch, I had forgotten to commit the symlink, it should be clickable as intended now

..    .. grid-item-card:: API Reference
..       :class-card: sd-card
..       :link: api
..       :link-type: doc

..       **Information-oriented** technical specs covering function definitions, parameters, return types, and class structures.

.. .. toctree::
..    :hidden:

..    getting_started
..    usage
..    api

-----

Partner with us
===============

.. div:: mb-4

   Whether you are a motivated student looking for opportunities in computational research, or an institution interested in funding parallel computing infrastructure, we invite you to collaborate with the Heat core development team.

.. tab-set::

   .. tab-item:: Student Projects
      :sync: student

      .. grid:: 1 1 2 2
         :gutter: 3
         :padding: 0
         :class-container: text-left

         .. grid-item::
            :columns: 12 12 7 7

            **Kickstart Your Research in HPC & AI**

            We are actively looking for motivated BSc, MSc, and student workers to tackle open challenges in memory-distributed data science.

            * **Core Topic Tracks:** Massively parallel tensor operations, communication backends, algorithm development, user-requested features.

>             * **What We Provide:** Direct mentorship from core open-source maintainers, computing time on top-tier HPC cluster environments, and clear paths to academic publication.

Reviewer: ```suggestion
            * **What We Provide:** Direct mentorship from the core maintainers, computing time on top-tier HPC cluster environments, and clear paths to academic publication.
```

         .. grid-item::
            :columns: 12 12 5 5
            :class: text-center

            .. div:: mt-4

               .. button-link:: https://github.com/helmholtz-analytics/heat/discussions/categories/student-projects
                  :color: primary
                  :class: sd-btn-primary sd-btn-block mb-2

                  Explore Open Theses & Issues

               .. button-link:: mailto:contact@example.com
                  :color: secondary
                  :outline:
                  :class: sd-btn-block

>                   Contact Core Maintainers

Reviewer: We should probably change this to a button to open a GitHub issue

Author: I removed this section entirely and all contact is via github issues

            We partner with both scientific institutions and industrial application partners during the proposal stage. If the project is successful, our core engineering team provides **ad-hoc support** to integrate Heat's distributed tensor operations directly into the project's pipeline.

         .. grid-item::
            :columns: 12 12 6 6

            **How We Integrate Into Project Consortia:**

            * **Technical Work Packages:** We help design dedicated work packages focused on software scaling, parallel acceleration, and performance optimization.
            * **HPC Middleware Expertise:** We act as the foundational parallel computing layer, helping translate domain-specific algorithms into cluster-ready architectures.
            * **Transfer Use-Case Co-Design:** We work side-by-side with your researchers or R&D engineers to implement custom distributed tensor operations unique to your scientific or technical discipline.

-----

Latest news
===========

.. grid:: 1 1 3 3
   :gutter: 3
   :class-container: mt-4

>    .. grid-item-card:: Save the Date

Reviewer: ```suggestion
   .. grid-item-card:: Workshop registration open
```

   .. grid-item-card:: HeatHub
      :class-card: sd-card
      :link: https://www.fz-juelich.de/en/rse/the_latest/congratulations-scienceserve-recipients

      .. image:: _static/images/ScienceServe_banner.png
         :alt: ScienceServe Funding Banner
         :align: center

      ^^^

      *October 2025*

      Our project HeatHub has been awarded Helmholtz ScienceServe funding for 2026!

Roadmap
=======

.. div:: mb-4

.. grid:: 1 1 3 3
   :gutter: 3
   :class-container: mb-2

   .. grid-item-card:: Active Milestones
      :class-card: sd-card
      :link: https://github.com/helmholtz-analytics/heat/milestones
      :text-align: center

      Track our progress toward upcoming minor and major framework releases.

   .. grid-item-card:: Live Project Board
      :class-card: sd-card
      :link: https://github.com/orgs/helmholtz-analytics/projects/13/views/6
      :text-align: center

      See active pull requests, bug fixes, and feature implementations in real time.

   .. grid-item-card:: Contribute Code
      :class-card: sd-card
      :link: https://github.com/helmholtz-analytics/heat/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22
      :text-align: center

>       Explore "good first issues" and feature requests for new contributors.

Reviewer: I would probably remove this section. The millstones and project board are too in depth for the landing page in mu opinion.

Author: Removed
